### PR TITLE
chore: remove deprecated fields in NSTemplateTier

### PR DESCRIPTION
**Pull request by xcoulon**
Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: pkg/test/masteruserrecord/assertion.go [hunk 1]
```diff
@@ -156,36 +156,32 @@ func (a *Assertion) AllUserAccountsHaveTier(tier toolchainv1alpha1.NSTemplateTie
...
+			} else {
+				assert.Equal(a.t, tier.Spec.ClusterResources.TemplateRef, ua.Spec.NSTemplateSet.ClusterResources.TemplateRef)
+			}
 		}
 	}
 	return a
 }
 
 func (a *Assertion) userAccountHasTier(userAccount toolchainv1alpha1.UserAccountEmbedded, tier toolchainv1alpha1.NSTemplateTier) {
-	assert.Equal(a.t, tier.Name, userAccount.Spec.NSTemplateSet.TierName)
```
Comment (alexeykazakov) [inline pkg/test/masteruserrecord/assertion.go:168]: Any reason to keep this function which doesn’t do anything anymore?

Comment (xcoulon) [reply pkg/test/masteruserrecord/assertion.go:168]: oh, let me fix that! 🤦‍♂️

Comment (xcoulon) [reply pkg/test/masteruserrecord/assertion.go:168]: done in 2bc10e7

File: pkg/cluster/cache.go [hunk 2]
```diff
@@ -56,13 +56,13 @@ func (c *kubeFedClusterClients) deleteFedCluster(name string) {
...
 	defer c.RUnlock()
-	cluster, ok := c.clusters[name]
+	_, ok := c.clusters[name]
 	if !ok && c.refreshCache != nil {
 		c.RUnlock()
 		c.refreshCache()
 		c.RLock()
 	}
-	cluster, ok = c.clusters[name]
+	cluster, ok := c.clusters[name]
```
Comment (xcoulon) [inline pkg/cluster/cache.go:65]: [out-of-scope for this PR so I can address that separately]
This change was suggested by golang-ci linter.

File: go.mod [hunk 3]
```diff
@@ -66,4 +66,6 @@ replace (
 	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20200204173128-addea2498afe // avoids case-insensitive import collision: "github.com/googleapis/gnostic/openapiv2" and "github.com/googleapis/gnostic/OpenAPIv2"
 )
 
+replace github.com/codeready-toolchain/api => ../api
```
Comment (MatousJobanek) [inline go.mod:69]: just a reminder :wink:

Comment (xcoulon) [reply go.mod]: done in 3d793c6

**Pull request merged**